src/components/grid-list/tile-coordinator.ts

```diff
@@ -0,0 +1,138 @@
+import {MdGridTile} from './grid-tile';
+import {MdGridTileTooWideError} from './grid-list-errors';
+
+/**
+ * Class for determining, from a list of tiles, the (row, col) position of each of those tiles
+ * in the grid. This is necessary (rather than just rendering the tiles in normal document flow)
+ * because the tiles can have a rowspan.
+ *
+ * The positioning algorithm greedily places each tile as soon as it encounters a gap in the grid
+ * large enough to accommodate it so that the tiles still render in the same order in which they
+ * are given.
+ *
+ * The basis of the algorithm is the use of an array to track the already placed tiles. Each
+ * element of the array corresponds to a column, and the value indicates how many cells in that
+ * column are already occupied; zero indicates an empty cell. Moving "down" to the next row
+ * decrements each value in the tracking array (indicating that the column is one cell closer to
+ * being free).
+ */
+export class TileCoordinator {
+  /** Tracking array (see class description). */
+  tracker: number[];
+
+  /** Index at which the search for the next gap will start. */
+  columnIndex: number = 0;
+
+  /** The current row index. */
+  rowIndex: number = 0;
+
+  /** Gets the total number of rows occupied by tiles */
+  get rowCount(): number { return this.rowIndex + 1; }
+
+  /** Gets the total span of rows occupied by tiles
+   ** Ex: a list with 1 row that contains a tile with rowspan 2 will have a total rowspan of 2 */
```

Comment:
Typical JsDoc format is only one `*` on each comment line after the first, with the `*/` going on its own line.